### PR TITLE
questionnaireresponse

### PR DESCRIPTION
**Pull request by apeirats-va**
Initial implementation of FHIR resource for QuestionnaireResponse.

### CODE REVIEW

File: us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/QuestionnaireResponse.java [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+  @Valid List<Extension> modifierExtension;
+
+  @Valid Identifier identifier;
+
+  @Valid List<Reference> basedOn;
+
+  @Valid List<Reference> partOf;
+
+  @Pattern(regexp = Fhir.CANONICAL)
+  @Valid
```
Comment (jblefkove-va) [inline us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/QuestionnaireResponse.java:79]: don't need `@Valid`

File: us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/QuestionnaireResponse.java [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+    stopped
+  }
+
+  @Data
+  @Builder
+  @Schema(name = "QuestionnaireResponseItemAnswer")
+  @AllArgsConstructor
+  @NoArgsConstructor(access = AccessLevel.PRIVATE)
+  @JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+  @ExactlyOneOf(
```
Comment (jblefkove-va) [inline us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/QuestionnaireResponse.java:113]: value is optional, so this is `AtMostOneOf`

File: us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/QuestionnaireResponse.java [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+        "valueDateTime",
+        "valueTime",
+        "valueString",
+        "valueUri",
+        "valueAttachment",
+        "valueCoding",
+        "valueQuantity",
+        "valueReference"
+      },
+      message = "Exactly one answer field must be specified")
```
Comment (jblefkove-va) [inline us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/QuestionnaireResponse.java:128]: value, not answer

File: us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/QuestionnaireResponse.java (hunk 1 above)
Comment (apeirats-va) [reply us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/QuestionnaireResponse.java:79]: how'd that get there

File: us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/QuestionnaireResponse.java (hunk 2 above)
Comment (apeirats-va) [reply us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/QuestionnaireResponse.java:113]: Looks like `ZeroOrOneOf` is what you meant.

**Review by jblefkove-va — APPROVED** [review]

**Pull request merged**